beetsplug/missing.py

```diff
@@ -106,35 +114,98 @@ def __init__(self):
+            helper(lib, args)
 
-            # Default format string for count mode.
-            if count:
-                fmt += ': $missing'
+        self._command.func = _miss
+        return [self._command]
 
-            for album in albums:
-                if count:
-                    if _missing_count(album):
-                        print_(format(album, fmt))
+    def _missing_tracks(self, lib, args):
+        """Logic for determining missing tracks from each album in the library
+        (the usual case)
+        """
+        albums = lib.albums(decargs(args))
 
-                else:
-                    for item in self._missing(album):
-                        print_(format(item, fmt))
+        count = self.config['count'].get()
+        total = self.config['total'].get()
+        fmt = config['format_album' if count else 'format_item'].get()
 
-        self._command.func = _miss
-        return [self._command]
+        if total:
+            print(sum([_missing_count(a) for a in albums]))
+            return
+
+        # Default format string for count mode.
+        if count:
+            fmt += ': $missing'
+
+        for album in albums:
+            if count:
+                if _missing_count(album):
+                    print_(format(album, fmt))
+
+            else:
+                for item in self._missing(album):
+                    print_(format(item, fmt))
+
+    def _missing_albums(self, lib, args):
+        """Logic for determining missing albums from each artist in the library
+        (-a case)
+        """
+        albums = lib.albums(decargs(args))
+        # build dict mapping artist to list of their albums in library
+        albums_by_artist = {}
+        for alb in albums:
+            artist = (alb['albumartist'], alb['mb_albumartistid'])
+            artistalbs = albums_by_artist.setdefault(artist, [])
+            artistalbs.append(alb)
+            albums_by_artist[artist] = artistalbs
+
+        # build dict mapping artist to list of all albums
+        m.set_useragent('beets', beets.__version__, 'http://beets.io/')
+
+        for artist, albums in albums_by_artist.items():
+            if artist[1] is None or artist[1] == "":
+                print_("[!] No musicbrainz ID for artist {}, skipping."
+                       .format(artist[0]), stream=sys.stderr)
+                continue
+
+            try:
+                releases = m.browse_releases(artist=artist[1])['release-list']
+            except MusicBrainzError as err:
+                print_("[!] Couldn't fetch info for artist '{}' ({}) - '{}'"
+                       .format(*artist, err), stream=sys.stderr)
+                continue
+
+            missing = []
+            present = []
+            for release in releases:
+                missing.append(release)
+                for alb in albums:
+                    # compare by title, don't care about specific releases
```

Comment:
good idea, was looking for something better than title but not super familiar with mb api -- will implement